### DISCUSSION TITLE
Green screen on camera

### DISCUSSION THREAD

**Discussion by khyati18** [google/filament#6685] [Q&A]
A green screen show ups intermittently. This is happening only with depth api supported devices and is happening even when depth mode is disabled. Using the latest filament version: 1.32.1

The camera texture is opaque, plane is texture transparent. I have already tried priority workarounds and material updates. And when depth mode is enable black screen shows up intermittently plus the green screen.

**Comment by romainguy** [comment]
What "depth mode" are you referring to?

**Reply by khyati18** [reply]
Config.DepthMode.DISABLED for camera session config

**Reply by romainguy** [reply]
That's not a Filament API

**Reply by khyati18** [reply]
With latest version of filament. I am using camera stream like this. The green screen issue is appearing. 

```
stream = Stream
    .Builder()
    .width(width)
    .height(height)
    .build(filament.engine)
Texture
    .Builder()
    .importTexture(cameraStreamTextureId.toLong())
    .sampler(Texture.Sampler.SAMPLER_EXTERNAL)
    .format(Texture.InternalFormat.RGB8)
    .build(filament.engine)
    .apply { setExternalStream(filament.engine, stream) }
```

With older version of filament. When I use it like this it works fine.
```
stream = Stream
    .Builder()
    .stream(cameraStreamTextureId.toLong())
    .width(width)
    .height(height)
    .build(filament.engine)
```

Then I am using this stream for the MaterialInstance. (setExternalStream(filament.engine, stream))

It is not ar-core issue seems. Ignore depth mode settings. @romainguy

**Comment by nischaysingla** [comment]
@khyati18 is your issue related to intermittent flickering green screen in place of camera texture?
If this is issue follow this!
https://github.com/google/filament/issues/5498 

Let me know if this solves your problem

**Reply by khyati18** [reply]
@nischaysingla I am still getting the green screen issue.

```
val cameraTextureIds = IntArray(4) { createExternalTextureId() }
session.setCameraTextureNames(cameraTextureIds) 
```
